src/highdicom/legacy/sop.py

```diff
@@ -25,384 +31,506 @@
-        ]
+        """
+        if not isinstance(tg, BaseTag):
+            tg = Tag(tg)
+        return '(0x{:0>4x}, 0x{:0>4x})'.format(tg.group, tg.element)
 
-        # Irradiation Event Identification (C) - CT/PET only
-        try:
-            irradiation_event_id_item = Dataset()
-            irradiation_event_id_item.IrradiationEventUID = \
-                ref_ds.IrradiationEventUID
-            perframe_item.IrradiationEventIdentificationSequence = [
-                irradiation_event_id_item,
-            ]
-        except AttributeError:
-            pass
-
-        # Temporal Position (U)
-        try:
-            temporal_position_item = Dataset()
-            temporal_position_item.TemporalPositionTimeOffset = \
-                ref_ds.TemporalPositionTimeOffset
-            perframe_item.TemporalPositionSequence = [
-                temporal_position_item,
-            ]
-        except AttributeError:
-            pass
+    @classmethod
+    def tag2kwstr(cls, tg: BaseTag) -> str:
+        """Converts tag to keyword and (group, element) form
+
+        """
+        return '{}-{:32.32s}'.format(
+            cls.tag2str(tg), keyword_for_tag(tg))
 
-        # Cardiac Synchronization (U)
-        # TODO: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.16.2.html#sect_C.7.6.16.2.7  # noqa
 
-        # Contrast/Bolus Usage (U) - MR/CT only
-        # TODO: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.16.2.html#sect_C.7.6.16.2.12  # noqa
+class GeometryOfSlice:
 
-        # Respiratory Synchronization (U)
-        # TODO: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.16.2.html#sect_C.7.6.16.2.17  # noqa
+    """A class for checking dicom slices geomtery/parallelization"""
 
-        # Real World Value Mapping (U) - PET only
-        # TODO: http://dicom.nema.org/medical/dicom/current/output/chtml/part03/sect_C.7.6.16.2.html#sect_C.7.6.16.2.11  # noqa
+    def __init__(self,
+                 row_vector: ndarray,
+                 col_vector: ndarray,
+                 top_left_corner_pos: ndarray,
+                 voxel_spaceing: ndarray,
+                 dimensions: tuple) -> None:
+        """
+        Parameters
+        ----------
+        row_vector: ndarray
+            3D vector representing row of the input slice
+        col_vector: ndarray
+            3D vector representing column the input  aslice
+        top_left_corner_pos: ndarray
+            3D point representing top left coner position of the input slice
+        voxel_spaceing: ndarray
+            Three element array. 1st and 2nd copied from PexelSpacing and the
+            3rd copied from SliceThickness
+        dimensions: tuple
+            3 element tuple holding x as number of cols, y as number of rows
+            and z as 1
+        """
+        self.RowVector = row_vector
+        self.ColVector = col_vector
+        self.TopLeftCornerPosition = top_left_corner_pos
```

Comment:
Maybe "origin" is a better name to be used here?

Also, as noted earlier, CamelCase is reserved for DICOM keywords.

Reply:
Sure, corrected. But the naming convention is copied from David's code.